jax/_src/array.py

```diff
@@ -680,54 +680,85 @@ def make_array_from_callback(
 def make_array_from_single_device_arrays(
     shape: Shape, sharding: Sharding, arrays: Sequence[basearray.Array]
 ) -> ArrayImpl:
-  r"""Returns a ``jax.Array`` from a sequence of ``jax.Array``\s on a single device.
+  r"""Returns a ``jax.Array`` from a sequence of ``jax.Array``\s each on an individual device.
 
-  You can use this function if you have already ``jax.device_put`` the value on
-  a single device and want to create a global Array. The smaller ``jax.Array``\s should be
-  addressable and belong to the current process.
+  To use this function, every device in the global mesh must have an addressable array that
+  is already on on-device, typically either via ``jax.device_put`` or generated directly onto the device.
+
+  The ``jax.Array``\s forming the input argument ``arrays`` must be local to the calling
+  process.
 
   Args:
-    shape : Shape of the ``jax.Array``.
-    sharding: A ``Sharding`` instance which describes how the ``jax.Array`` is
-      laid out across devices.
-    arrays: Sequence of ``jax.Array``\s that are on a single device.
+    shape : Shape of the output ``jax.Array``. This is redundant with ``sharding`` and ``arrays`` and serves as a
+      double check.
+    sharding: A ``Sharding`` instance which describes how the output ``jax.Array`` is
+      shaped and replicated.
+    arrays: Sequence of ``jax.Array``\s that are each on a single device for this process. ``len(arrays)``
+      must equal ``len(jax.local_devices())`` and the shape of each array must be the same. For multiprocess code,
```

Comment:
The multiprocess part of this sentence is very specific.

You can say that in the example you have below but not here. So please remove that.